bliss/simulator/decoder.py

```diff
@@ -151,25 +154,26 @@ def render_images(self, tile_cat: TileCatalog, image_ids):
             psf = psfs[b]
             shift, wcs = self.pixel_shift()
             wcs_batch.append(wcs)
-            for band in range(self.n_bands):
-                gs_img = galsim.Image(array=images[b, band], scale=self.pixel_scale)
-                for s in range(n_sources):
-                    source_params = full_cat.one_source(b, s)
-                    source_type = source_params["source_type"].item()
-                    renderer = self.source_renderers[source_type]  # NOTE: SDSS-Specific!
-                    galsim_obj = renderer(psf, band, source_params)
-                    plocs0, plocs1 = source_params["plocs"]
-                    offset = np.array([plocs1 - (slen_w / 2), plocs0 - (slen_h / 2)])
-                    offset += shift[band]
-
-                    # essentially all the runtime of the simulator is incurred by this call
-                    # to drawImage
-                    galsim_obj.drawImage(
-                        offset=offset,
-                        method=self.psf_draw_method,
-                        add_to_image=True,
-                        image=gs_img,
-                    )
+            for d in range(coadd_depth):
+                for band in range(self.n_bands):
+                    gs_img = galsim.Image(array=images[b, d, band], scale=self.pixel_scale)
+                    for s in range(n_sources):
+                        source_params = full_cat.one_source(b, s)
+                        source_type = source_params["source_type"].item()
+                        renderer = self.source_renderers[source_type]
+                        galsim_obj = renderer(psf, band, source_params)
+                        plocs0, plocs1 = source_params["plocs"]
+                        offset = np.array([plocs1 - (slen_w / 2), plocs0 - (slen_h / 2)])
+                        offset += shift[band]
```

Comment:
Don't you want to apply a different shift here to each raw image `d`? Currently it looks like a particular band (say, the `r` band) will have the same shift in all the raw images.

Reply:
Ah yes, the shifts will probably need to be indexed by `[band, d(epth)]`

Reply:
> Isn't the point of `for d in range(coadd_depth)` loop to generate raw images that differ in some way? (What am I missing here?)

For ease of implementation, this manual dithering via WCS offset + variation due to prior sampling might suffice; I didn't think making the image generation even less synthetic, via something like inferring variations from different exposure numbers, to be viable / worth it. But I might be misunderstanding this whole process completely.